[DOCU-3188] OSS Client Portal (#5631)
Kong/docs.konghq.com: app/konnect/dev-portal/customization/self-hosted-portal.md

---
title: About Self-Hosted Dev Portal
content_type: explanation
badge: oss
---

> In {{site.konnect_saas}}, you have two hosting options for your Dev Portal: a cloud hosted Dev Portal with {{site.konnect_short_name}} or a self-hosted, open source Dev Portal. With the cloud hosted Dev Portal in {{site.konnect_short_name}}, your Dev Portal is hosted for you and it simplifies your deployment experience. The self-hosted, open source Dev Portal provides all the same features as the cloud-hosted portal, it just also gives you control over your hosting service and where your Dev Portal is hosted. 

Reviewer: ```suggestion
In {{site.konnect_saas}}, you have two hosting options for your Dev Portal: a cloud hosted Dev Portal with {{site.konnect_short_name}} or a self-hosted, open source Dev Portal. 
* **Cloud hosted Dev Portal in {{site.konnect_short_name}}**: Your Dev Portal is hosted for you and it simplifies your deployment experience. 
* **Self-hosted, open source Dev Portal**: Provides all the same features as the cloud-hosted portal, but also gives you control over your hosting service and where your Dev Portal is hosted. 
```
Splitting this paragraph into a list to make it easier to skim.

You can use the open source Dev Portal to display your APIs to developers on a self-hosted website. This page explains the benefits of using a self-hosted Dev Portal, a high-level overview of how the self-hosted portal works, and how to enable it.

## Self-hosted Dev Portal benefits

There are several benefits to keep in mind when deciding whether to use a {{site.konnect_short_name}}-hosted or self-hosted Dev Portal. The self-hosted portal provides the following benefits: 

* **Fully customizable:** Use the [example frontend Dev Portal application](https://github.com/Kong/konnect-portal) as a starting point and then customize Dev Portal for your needs using the Portal API. You can also integrate the API specs with workflows tailored to your organization's own processes.
* **Hosting service choice:** When you self-host, you also get to choose which hosting service you use to deploy your Dev Portal. 

> * **Range of customization options:** With the self-hosted Dev Portal, you determine how much you want to customize. You can choose to use the [example application](https://github.com/Kong/konnect-portal) right out of the box or you can use the Portal API to have more fine-grained control.

Reviewer 2: ```suggestion
* **Range of customization options:** With the self-hosted Dev Portal, you determine how much you want to customize. You can choose to use the example application right out of the box or you can use the Portal API to have more fine-grained control.
```
Feels odd to link this a second time with a different name because as the reader I felt like I was going to be taken to a different link -- I already had the repo tab open from the first bullet's link. You can either remove the link, or keep the link but call it "example front end dev portal application" like you did in the first bullet.

> * **Range of customization options:** With the self-hosted Dev Portal, you determine how much you want to customize. You can choose to use the [example application](https://github.com/Kong/konnect-portal) right out of the box or you can use the Portal API to have more fine-grained control.

Reviewer: ```suggestion
* **Range of customization options:** With the self-hosted Dev Portal, you determine how much you want to customize. You can choose to use the [example application](https://github.com/Kong/konnect-portal) right out of the box, or use the Portal API for more fine-grained control.
```

## How the self-hosted Dev Portal works 

The following diagram explains how the self-hosted Dev Portal works by using the example frontend application:

> ![DIAGRAM HERE](/assets/images/docs/konnect/diagram.png)

Reviewer 2: Diagram doesn't work
![image](https://github.com/Kong/docs.konghq.com/assets/23319190/a9cc9585-93b2-42d0-89e2-5e646d05d7f6)

> Figure 1: Diagram that shows how the example frontend application communicates with the developer's browser and {{site.konnect_short_name}}. The self-hosted Dev Portal is hosted in your infrastructure and the application is served to the developer browser. The developer browser communicates with the Portal API for API requests and {{site.konnect_short_name}} supplies the corresponding content. 

 
## Enable a self-hosted Dev Portal

> To enable the self-hosted Dev Portal in {{site.konnect_short_name}}, click {% konnect_icon dev-portal %} **Dev Portal** > [**Settings**](https://cloud.konghq.com/portal/portal-settings) in the sidebar and then click the **Portal Domain** tab. Once you specify a custom portal domain API in the **Custom Hosted Domain** field and a custom portal client domain in the **Custom Self-Hosted UI Domain** field, your self-hosted portal is enabled. After enabling the portal, you must configure the settings in your static site generator to deploy your Dev Portal.

Reviewer: ```suggestion
You can enable the self-hosted Dev Portal through {% konnect_icon dev-portal %} **Dev Portal** > [**Settings**](https://cloud.konghq.com/portal/portal-settings), then setting up a custom domain in the **Portal Domain** tab. 

The following fields are required:
* Custom Hosted Domain
* Custom Self-Hosted UI Domain

After enabling the portal, you must configure the settings in your static site generator to deploy your Dev Portal.
```
Hard to follow the instructions in a paragraph.

---
title: About Self-Hosted Dev Portal
content_type: explanation

> badge: oss

Author: @rosskukulinski So the OSS portal client = example application (konnect-portal repo)? Can you just use the OSS portal client and enable the settings in the Konnect UI and do everything with open source (as long as you don't plan to use the APIs)?

> badge: oss

Reviewer 3: I think we should just remove the badge entirely.  We changed course and are just making the Self-hosted dev portal available to all Konnect tiers.

But the self-hosted dev portal, while OSS, _will not do anything_ unless it's tied to a Konnect organization. Hence I don't think the OSS badge makes a ton of sense?

> badge: oss

Author: Sounds good, I'll just remove the badge entirely!

---

> In {{site.konnect_saas}}, you have two hosting options for your Dev Portal: a cloud hosted Dev Portal with {{site.konnect_short_name}} or a self-hosted, open source Dev Portal. With the cloud hosted Dev Portal in {{site.konnect_short_name}}, your Dev Portal is hosted for you and it simplifies your deployment experience. The self-hosted, open source Dev Portal provides all the same features as the cloud-hosted portal, it just also gives you control over your hosting service and where your Dev Portal is hosted. 

Reviewer 4: ```suggestion
In {{site.konnect_saas}}, you have two out-of-the box solutions for your Dev Portal: a cloud-hosted Dev Portal with {{site.konnect_short_name}} or a self-hosted, open source Dev Portal. With the cloud-hosted Dev Portal in {{site.konnect_short_name}}, your Dev Portal is hosted for you and it simplifies your deployment experience. The self-hosted, open source Dev Portal provides all the same features as the cloud-hosted portal, it just also gives you control over your hosting service and where your Dev Portal is hosted. 
```

## How the self-hosted Dev Portal works 

The following diagram explains how the self-hosted Dev Portal works by using the example frontend application:

![DIAGRAM HERE](/assets/images/docs/konnect/diagram.png)

> Figure 1: Diagram that shows how the example frontend application communicates with the developer's browser and {{site.konnect_short_name}}. The self-hosted Dev Portal is hosted in your infrastructure and the application is served to the developer browser. The developer browser communicates with the Portal API for API requests and {{site.konnect_short_name}} supplies the corresponding content. 

The Dev Portal API is always hosted by {{site.konnect_short_name}}, no matter if you use the self-hosted or {{site.konnect_short_name}}-hosted Dev Portal. The Dev Portal API maps to the Portal Domain (and optional Custom Hosted Domain, which is a CNAME). Depending on which domain settings you configure and if you are self-hosting or not, the Dev Portal UI domain will differ.

![DIAGRAM HERE](/assets/images/docs/konnect/diagram.png)

> > Figure 2: Diagram that shows that self-hosted and {{site.konnect_short_name}}-hosted Dev Portals use different UI domains, but both use the same {{site.konnect_short_name}}-hosted Dev Portal API.

Author: @Guaris If you have a chance, could you review this? I added lines 28-32 to accommodate a diagram Ross wanted to add. I hope that it makes sense even without the diagram added. Hopefully marketing wraps up those diagrams soon.

Author: Oh, and if that looks good (I also read through the whole thing again), could you approve this PR so I can hand it over to Ross?
